packages/extension/src/common/keyNavigationService.js

```diff
@@ -17,75 +17,56 @@ export const validKeys = {
   esc: 27,
 };
 
-function getNewPostEl(keyCode, currentElement, insaneMode) {
-  if (keyCode === validKeys.h && !insaneMode) return getAbovePost(currentElement);
+const validKeysValues = Object.values(validKeys);
 
-  if (keyCode === validKeys.l && !insaneMode) return getBelowPost(currentElement);
+function navigateFeed(keyCode, currentElement, insaneMode) {
+  if (keyCode === validKeys.h && !insaneMode) return getPreviousPost(currentElement);
 
-  if (keyCode === validKeys.j && insaneMode) return getBelowPost(currentElement);
+  if (keyCode === validKeys.l && !insaneMode) return getNextPost(currentElement);
 
-  if (keyCode === validKeys.k && insaneMode) return getAbovePost(currentElement);
+  if (keyCode === validKeys.j) return getBelowPost(currentElement, insaneMode);
+
+  if (keyCode === validKeys.k) return getAbovePost(currentElement, insaneMode);
 
   return currentElement;
 }
 
-function triggerBookmark(post) {
-  if (post.ad) return null;
+function triggerBookmark(article) {
+  if (article.ad) return null;
 
-  return post.$emit('bookmark', { post: post.post, bookmarked: !post.post.bookmarked });
+  return article.$emit('bookmark', { post: article.post, bookmarked: !article.post.bookmarked });
 }
 
-function getCurrentPost(posts, current) {
-  if (!current) return null;
-
-  const postOrAdProp = current.post || current.ad;
+function navigateDaily({ keyCode }) {
+  if (validKeysValues.indexOf(keyCode) === -1) return null;
 
-  return posts.find(article => [article.ad, article.post].indexOf(postOrAdProp) !== -1);
-}
-
-function navigate(keyCode, posts, search, backToMainFeed, { current, insaneMode }) {
-  if (keyCode === validKeys.esc) return backToMainFeed();
+  const { daFeedRef, hoveredPost } = store.state.feed;
 
-  if (posts.length === 0) return null;
+  const ref = daFeedRef();
 
-  const item = getCurrentPost(posts, current);
+  if (keyCode === validKeys.esc) return store.dispatch('feed/backToMainFeed');
```

Comment:
I think here we should also reset the "show bookmarks" state. I often click `esc` while browsing my bookmarks and it just refreshes the feed instead of going back to the main feed. What do you think?

Reply:
Will definitely add this on the list! 💯